moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/current_state_monitor.h

```diff
@@ -168,10 +168,30 @@ class CurrentStateMonitor
    *  @return Returns the map from joint names to joint state values*/
   std::map<std::string, double> getCurrentStateValues() const;
 
-  /** @brief Wait for at most \e wait_time seconds (default 1s) for a robot state more recent than t
+  /** @brief Wait for at most \e wait_time seconds (default 1 s) for a robot state more recent than \e t
+   *  @return true on success, false if an up-to-date robot state message wasn't received within \e wait_time
+   */
+  inline bool waitForCurrentState(const ros::Time t = ros::Time::now(), double wait_time = 1.0) const
```

Comment:
MoveIt is sprinkled with these throughout the code base, and we don't have a clear policy for this.
Personally I prefer to have the `inline` there to make it clear that no symbol will be generated by the compiler.
But I see the benefits of having one modifier less in the code...

Reply:
It's not like I heavily care either way ... I looked at `RobotState` for reference

Reply:
yeah, as I said, it's inconsistent. But it also looks like there is no clang-format option for it. So I don't really care either :shrug:

Reply:
That makes three of us. I'll leave this as is, fix the other two issues later this week and push an update.

Reply:
Copypasting the salient part from here, emphasis mine [(https://en.cppreference.com/w/cpp/language/inline](https://en.cppreference.com/w/cpp/language/inline))

>The original intent of the inline keyword was to serve as an indicator to the optimizer that inline substitution of a function is preferred over function call, that is, instead of executing the function call CPU instruction to transfer control to the function body, a copy of the function body is executed without generating the call. This avoids overhead created by the function call (passing the arguments and retrieving the result) but it may result in a larger executable as the code for the function has to be repeated multiple times.

>Since this meaning of the keyword inline is non-binding, **compilers are free to use inline substitution for any function that's not marked inline, and are free to generate function calls to any function marked inline.** Those optimization choices do not change the rules regarding multiple definitions and shared statics listed above. 

Given how `inline` doesn't actually guarantee any inlining (C++ :heart:), and actually nowadays means (again copypasting)
>Because the meaning of the keyword inline for functions came to mean "multiple definitions are permitted" rather than "inlining is preferred", that meaning was extended to variables. (since C++17)

Based on those couple paragraphs, I'd say `inline` doesn't apply here and should be therefore removed.